db/data_migration/20170522134843_convert_wlna_to_news_articles.rb

```diff
@@ -0,0 +1,39 @@
+def update_db_records_to_news_article(editions)
+  editions.update_all(
+    type: "NewsArticle",
+    news_article_type_id: NewsArticleType::WorldNewsStory.id
+  )
+end
+
+def update_rummager(editions)
+  editions.each(&:save)
+end
+
+def editions_including_deleted(document_id)
+  Edition.unscoped.where(document_id: document_id)
+end
+
+world_location_news_articles = WorldLocationNewsArticle.all
+wlna_documents = Document.where(id: world_location_news_articles.pluck(:document_id).uniq)
+
+puts "Migrating #{wlna_documents.count} WLNA -> NewsArticle"
+
+wlna_documents.each do |wlna_document|
+  wlna_document.document_type = "NewsArticle"
+  #updating sluggable_string will regenerate the slug and  de-duplicate
+  #if there is already a news article with this slug
+  begin
+    wlna_document.update_attributes(sluggable_string: wlna_document.slug)
+  rescue ActiveRecord::RecordNotUnique
+    puts "NotUnique: #{wlna_document.id}"
```

Comment:
The `ActiveRecord::RecordNotUnique` error is what was being thrown originally when we attempted to change the `document_type` on a `Document` that had a duplicate slug. I caught it here initially so as the migration continued to run and then I was left on subsequent runs in development with only WLNA that had duplicate slugs. When we run this now it should never happen but I left it in to ensure that we don't leave the DB half migrated. It will just leave the 'offenders' which we can address individually. The slug does get changed on the object but it doesn't get persisted as, if this error is raised, it has violated the unique key constraint on the database.